### PR TITLE
Replaces istypes with the proper tool_behaviour checks.

### LINKED ISSUE #40410: Airlocks use static types for tool interaction instead of the tool_behaviour system
https://github.com/tgstation/tgstation/blob/master/code/game/machinery/doors/door.dm#L183

Which means they're not weldable with custom welders such as, currently, plasma cutters, and not crowbarable by anything but crowbars and fireaxes.

### PR DESCRIPTION
**Pull request by ShizCalev**
Resolves the second issue from #40409
Fixes #40410

:cl: ShizCalev
fix: Resolved a large number of instances where items would not work as their respective tool types (ie plasma cutters not working as welders.)
fix: Using an item that acts as a multitool that does not have a buffer on an object that requires a buffer will now consistently provide user feedback for why it failed.
/:cl:

### CODE REVIEW

**Comment by yoyobatty** [comment]
Does this fix mech chargers? Wiki description says you can link em via muktitool, wasn't sure if this was done because as is it looks for an adjacent charger when you hit reconnect on the computer

**Comment by ShizCalev** [comment]
@yoyobatty nope, there's no logic in the code associated with multitools for mech recharges. from what i can tell that's an instance of a planned feature making it's way onto the wiki years ago that was never actually added. would only take like 5 minutes to add it in another PR, however.

**Comment by Barhandar** [comment]
Why would you need to actively connect something entirely optional? (pretty sure it doesn't need the console to work) It's better to fix the wiki rather than change mech chargers to require linkup.

**Comment by yoyobatty** [comment]
The issue is that you can't reconnect mech chargers unless it's directly adjacent to the console. All maps have it so they're one apart so if you open the panel or disassemble it, it won't detect until you move it around

**Comment by Barhandar** [comment]
In my experience they were working just fine when built as a golem with one square of space between console and the charger machine (i.e. space for the mech). Logically that'll also happen if you modify the stationside one then stick a mech in it.

File: code/game/machinery/porta_turret/portable_turret.dm [hunk 1]
```diff
@@ -282,7 +282,9 @@
 			to_chat(user, "<span class='notice'>Controls are now [locked ? "locked" : "unlocked"].</span>")
 		else
 			to_chat(user, "<span class='notice'>Access denied.</span>")
-	else if(istype(I, /obj/item/multitool) && !locked)
+	else if(I.tool_behaviour == TOOL_MULTITOOL && !locked)
+		if(!multitool_check_buffer(user, I))
```
Comment (AnturK) [inline code/game/machinery/porta_turret/portable_turret.dm]: Shouldn't these checks be only where you need the buffer to be full ?

**Pull request merged**